R/assertions.R

```diff
@@ -1698,27 +1700,29 @@ assert_one_to_one <- function(dataset, vars1, vars2) {
 #'   dataset = my_data,
 #'   var = USUBJID
 #' ))
-assert_date_var <- function(dataset, var, dataset_name = NULL, var_name = NULL) {
+assert_date_var <- function(dataset,
+                            var,
+                            dataset_name = rlang::caller_arg(dataset),
+                            var_name = rlang::caller_arg(var),
+                            message = NULL,
+                            class = "assert_date_var",
+                            call = parent.frame()) {
   var <- assert_symbol(enexpr(var))
   assert_data_frame(dataset, required_vars = exprs(!!var))
-  assert_character_scalar(dataset_name, optional = TRUE)
-  assert_character_scalar(var_name, optional = TRUE)
+  assert_character_scalar(dataset_name)
+  assert_character_scalar(var_name)
   column <- pull(dataset, !!var)
-  if (is.null(dataset_name)) {
-    dataset_name <- arg_name(substitute(dataset))
-  }
-  if (is.null(var_name)) {
-    var_name <- as_label(var)
-  }
+
   if (!is.instant(column)) {
-    abort(paste0(
-      "`",
-      var_name,
-      "` in dataset `",
-      dataset_name,
-      "` is not a date or datetime variable but is ",
-      friendly_type_of(column)
-    ))
+    message <- message %||%
+      "Column {.val {var_name}} in data frame {.code {dataset_name}} must be
```

Comment:
Is there a way to get the name of the dataset object for the user? 
![image](https://github.com/pharmaverse/admiraldev/assets/10111024/f1aba492-0670-420a-900e-4b5c9a6bbf5c)

Reply:
@bms63 probably? I just implemented the same messaging in the previous iteration. I'll see what i can do

Reply:
@bms63 I think your request makes perfect sense in a simple situation like `mtcars |> foo()`: you would like to see `mtcars`.

But we can quickly get into some ambiguous situations:

```r
mtcars %>% foo()                        # here the data frame is named `.`
mtcars %>% filter(cyl == 6) %>% foo()   # the data frame is again named `.`
mtcars |> filter(cyl == 6) |> foo()     # here the data frame is named `filter(mtcars, cyl == 6)`
```

What do you think? Should we just stick with the argument name? Then it's always defined.

Reply:
Ah - yes.  keep it simple!